dolphin/drivers/dellemc_storage/vmax.py

```diff
@@ -0,0 +1,149 @@
+            username = register_info['username']
+            password = register_info['password']
+
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                u4v_version=u4v_version, server_ip=server_ip, port=port,
+                verify=verify, username=username, password=password)
+
+            # Get the Unisphere version
+            version = conn.common.get_uni_version()
+
+            # Retrieve a list of arrays managed by your instance of Unisphere
+            array_list = conn.common.get_array_list()
+
+            # Close the session
+            conn.close_session()
+
+        except:
+            print('Failed to connect to storage')
+
+        else:
+            # Update storage_id's that this driver manages
+            if VMAXStorageDriver.storage_ids:
+                if self.storage_id in VMAXStorageDriver.storage_ids:
+                    LOG.info('Storage is already registered!')
+                else:
+                    VMAXStorageDriver.storage_ids.append(self.storage_id)
+            else:
+                VMAXStorageDriver.storage_ids = [self.storage_id]
+        
+        print('register_storage()')
+        return VMAXStorageDriver.storage_ids
+
+    def get_storage(self, context):
+        try:
+            u4v_version = register_info['u4v_version']
+            server_ip = register_info['server_ip']
+            port = register_info['port']
+            verify = register_info['verify']
+            username = register_info['username']
+            password = register_info['password']
+
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                u4v_version=u4v_version, server_ip=server_ip, port=port,
+                verify=verify, username=username, password=password)
+
+
+            # Get the Unisphere version
+            version = conn.common.get_uni_version()
+
+            # Retrieve a list of arrays managed by your instance of Unisphere
+            array_list = conn.common.get_array_list()
+
+            # Output results to screen
+            LOG.info('version is: {ver}'.format(ver=version[0]))
+            LOG.info('This instance of Unisphere instance manages the following arrays: '
+                '{arr_list}'.format(arr_list=array_list))
+
+            # GET those arrays which are local to this instance of Unisphere
+            local_array_list = list()
+            for array_id in array_list:
+                array_details = conn.common.get_array(array_id)
+                if array_details['local']:
+                    local_array_list.append(array_id)
+
+            # Output results to screen
+            LOG.info('The following arrays are local to this Unisphere instance: '
+                '{arr_list}'.format(arr_list=local_array_list))
+
+            # Close the session
+            conn.close_session()
+
+        except:
+            print('Failed to connect to storage')
```

Comment:
same as above about exception

Reply:
This may be added later, as at preset it will not add any info.